inc/cleantalk-common.php

```diff
@@ -181,7 +181,7 @@ function apbct_base_call($params = array(), $reg_flag = false)
      */
     if (
         empty($params['sender_email']) ||
-        ( isset($params['exception_action']) && $default_params['exception_action'] == 1 )
+        ( isset($params['exception_action']) && $params['exception_action'] == 1 )
     ) {
         $default_params['exception_action'] = 1;
```

Comment:
Are you sure that we need this?
```
 $ct_request = new CleantalkRequest(
        \Cleantalk\ApbctWP\Helper::arrayMergeSaveNumericKeysRecursive($default_params, $params)
);
```
Because $default_paramswill be allways overwritten by $params in the code above.

Reply:
This is the old code with a "beard". We can implement new params in the future, so the merging of these arrays needs to be.

Reply:
I doubt that it's necessary. On line 186 are using $default_params which is wrong. We already have $params to alter, default is default, it should be static.

Just take a look at the condition. Pseudocode:
`if 
! $params['sender_email'] || $params['exception_action'] == 1
then
$default_params['exception_action'] = 1`
Does it make sense to you?

If we change default_params to $params, then we'll get this:
`if 
! $params['sender_email'] || $params['exception_action'] == 1
then
$params['exception_action'] = 1`
This has even less sense than the previous.

So it should be:
`if
! $params['sender_email']
then
$params['exception_action'] = 1`

Reply:
Up

Reply:
Agree. Done https://github.com/CleanTalk/wordpress-antispam/pull/74/commits/88888fa9024f4f4a3c175c91465fc718e0edf1a1